cmd/soci/commands/push.go

```diff
@@ -109,11 +116,30 @@ if they are available in the snapshotter's local content store.
 				return indexDescriptors[i].CreatedAt.Before(indexDescriptors[j].CreatedAt)
 			})
 
-			username := cliContext.String("user")
+			refspec, err := reference.Parse(ref)
+			if err != nil {
+				return err
+			}
+
+			var username string
 			var secret string
-			if i := strings.IndexByte(username, ':'); i > 0 {
-				secret = username[i+1:]
-				username = username[0:i]
+			var ac dockercliconfigtypes.AuthConfig
+			if cliContext.Bool("dockerauth") {
```

Comment:
I wonder if the default should be to just figure out credentials. For now, that would mean use docker credentials if `--user` isn't specified, but in the future maybe there would be other ways to authenticate.

It might still be useful to have this flag to be explicit about it and later add a debug flag which will log what credentials it's using.

What do you think? Is that too magical?

Reply:
I think that if we're going to be intentional about this decision we should probably mimic what nerdctl does, since I think it's the most common such tool that supports docker auth and isn't docker (where it's obviously implicit). I have not investigated nerdctl's behavior.

Reply:
I think it's worth taking a look. I'd rather not introduce a flag that we might immediately remove.

Reply:
It looks like nerdctl uses docker config automatically iff username and password aren't specified on the command line. I'll switch to that behavior.